### PR TITLE
Feat/dsm by date

### PR DESCRIPTION
**Pull request by ayusht965**
Added a date filter on DSM header. The user can toggle the date and the Celebrations, Announcements and Team Requests appears according to the date. The user cannot go to future dates.

Screenshot 2023-03-30 at 4 14 57 PM

Screenshot 2023-03-30 at 4 10 31 PM

Screenshot 2023-03-30 at 4 10 00 PM

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -31,10 +32,10 @@
     "react-csv": "^2.2.2",
     "react-datepicker": "^4.10.0",
     "react-dom": "^18.2.0",
-    "react-material-ui-carousel": "^3.4.2",
     "react-draft-wysiwyg": "^1.15.0",
     "react-grid-layout": "^1.3.4",
     "react-markdown": "^8.0.5",
+    "react-material-ui-carousel": "^3.4.2",
```
Comment (akshayumck) [inline package.json:38]: whats the change in here ?

File: src/components/dsm/Announcements.js [hunk 2]
```diff
@@ -129,9 +131,13 @@ export default function Announcements() {
           }}
         >
           <Typography variant="dsmSubMain">Announcements</Typography>
-          <IconButton onClick={handleAddButtonClick}>
-            <AddCircleIcon color="primary" />
-          </IconButton>
+          {
+            format(selectedDate, 'yyyy-MM-dd') === format(new Date(), 'yyyy-MM-dd') && (
```
Comment (akshayumck) [inline src/components/dsm/Announcements.js:135]: can you use the same format date that is added in utils ?

File: src/components/dsm/DSMHeader.js [hunk 3]
```diff
@@ -1,14 +1,45 @@
...
 
-export default function DSMHeader() {
   return (
     <Box>
       <Box sx={{ backgroundColor: 'primary.light', padding: '25px' }} />
       <AppBar position="static" sx={{ background: 'transparent', boxShadow: 'none' }} >
         <Container maxWidth="xl">
           <Toolbar disableGutters sx={{ display: 'flex', justifyContent: 'space-between' }}>
-            <Box>
+            <Box sx={{ display: 'flex', justifyContent: 'space-between', width: '100%' }}>
```
Comment (akshayumck) [inline src/components/dsm/DSMHeader.js:42]: use the same comment from karthi's search pr here also

**Comment by akshayumck** [comment]
please consider the comments in this pr and other pr's , everything else looks good to me

**Review by akshayumck — APPROVED** [review]

File: package.json (hunk 1 above)
Comment (ayusht965) [reply package.json:38]: just the order got changed

File: src/components/dsm/DSMHeader.js (hunk 3 above)
Comment (ayusht965) [reply src/components/dsm/DSMHeader.js]: fixed it

**Pull request merged**